pennylane/interfaces/torch.py

```diff
@@ -76,6 +76,74 @@ def forward(ctx, input_kwargs, *input_):
         if hasattr(res, "numpy"):
             res = res.numpy()
 
+        ctx.saved_grad_matrices = {}
+
+        def _evaluate_grad_matrix(grad_matrix_fn):
+            """Convenience function for generating gradient matrices
+            for the given parameter values.
+
+            This function serves two purposes:
+
+            * Avoids duplicating logic surrounding parameter unwrapping/wrapping
+
+            * Takes advantage of closure, to cache computed gradient matrices via
+              the ctx.saved_grad_matrices attribute, to avoid gradient matrices being
+              computed multiple redundant times.
+
+              This is particularly useful when differentiating vector-valued QNodes.
+              Because PyTorch requests the vector-GradMatrix product,
+              and *not* the full GradMatrix, differentiating vector-valued
+              functions will result in multiple backward passes.
+            """
+            if grad_matrix_fn in ctx.saved_grad_matrices:
+                return ctx.saved_grad_matrices[grad_matrix_fn]
+
+            tape.set_parameters(ctx.all_params_unwrapped, trainable_only=False)
+            grad_matrix = getattr(tape, grad_matrix_fn)(
+                device, params=ctx.args, **tape.jacobian_options

```

Comment:
This function receives `**tape.jacobian_options` for keywords, but it is not just used for the jacobian.  This line is also used for the hessian, in which circumstances we would need to pass `**tape.hessian_options`.
```suggestion
                device, params=ctx.args, **getattr(tape,grad_matrix_fn+"_options")
```

Reply:
Oh, this is actually intentional 🤔 Currently, the QNode uses the `jacobian_options` dictionary as a catch-all for _all_ differentiation options. I'm not too sure why `hessian_options` exists, as as far as I know it will always be empty

Reply:
Oh, I stuck `hessian_options` because I thought it would be needed.  I can add a suggestion to remove that.